tests/utils/test_db.py

```diff
@@ -251,3 +252,16 @@ def test_alembic_configuration(self):
         import airflow
 
         assert config.config_file_name == os.path.join(os.path.dirname(airflow.__file__), "alembic.ini")
+
+    def test_bool_lazy_select_sequence(self):
+        class MockSession:
+            def __init__(self):
+                pass
+
+            def scalar(self, stmt):
+                return None
+
+        t = Table("t", MetaData(), Column("id", Integer, primary_key=True))
+        lss = LazySelectSequence.from_select(select(t.c.id), order_by=[], session=MockSession())
+
+        assert not bool(lss)
```

Comment:
Will this fail in the original case?

```python
assert not bool(None)
```

seems to work fine

Reply:
Correct, `bool(None)` returns false. However, the problem is that prior to this change, `bool(lss)` returns `None`, which results in `TypeError: __bool__ should return bool, returned NoneType`, which I'm seeing in production.

Reply:
So really this is a case where either the assertion is true or an exception is raised. Perhaps there is a more explicit way to test this, or I could add a comment. Whatever you think is best!

Reply:
Ah, got it

I think `assert bool(lss) is False` would probably be better🤔

Reply:
Yeah I think you're right. It's much clearer